hybris/common/q/linker.cpp

```diff
@@ -3471,33 +3471,23 @@ bool soinfo::prelink_image() {
         break;
 
       case DT_HASH:
-        if(d->d_un.d_ptr > load_bias)
-          base = d->d_un.d_ptr;
-        else
-          base = load_bias + d->d_un.d_ptr;
-
-        nbucket_ = reinterpret_cast<uint32_t*>(base)[0];
-        nchain_ = reinterpret_cast<uint32_t*>(base)[1];
-        bucket_ = reinterpret_cast<uint32_t*>(base + 8);
-        chain_ = reinterpret_cast<uint32_t*>(base+ 8 + nbucket_ * 4);
+        nbucket_ = reinterpret_cast<uint32_t*>(load_bias + d->d_un.d_ptr)[0];
```

Comment:
These lines here are changed in both this and previous commit and here it mostly reverting previous those changes, maybe check what are the actual changes and do the changes in one of the commits.

Reply:
The following hunks eliminated each other after squashing:
https://github.com/libhybris/libhybris/pull/467/commits/bafeaea3876405d9e9a6a9dc606fd1ea86cab74a#diff-5b410cf47d1a8c88895533ca37eda37483a5c32da2b45cfc206c520cdea132d3L3453
https://github.com/libhybris/libhybris/pull/467/commits/d2c1d23772a062304df7246536975a69cdfd4643#diff-5b410cf47d1a8c88895533ca37eda37483a5c32da2b45cfc206c520cdea132d3L3437
https://github.com/libhybris/libhybris/pull/467/commits/9d72ca77e672675a459d2eb8fe16ad053cbdb654#diff-5b410cf47d1a8c88895533ca37eda37483a5c32da2b45cfc206c520cdea132d3L3474

This is how it looks after squashing, and the contents of linker.cpp are exactly the same as in this PR: https://github.com/sledges/libhybris-upstream/blame/q-support-squashed-buckets/hybris/common/q/linker.cpp#L3472 (will update this PR after fixing other bits).

Reply:
Pushed the cleaned (=squashed) version.